### PR TITLE
Upload benchmarks on pull request merge

### PR DESCRIPTION
**Pull request by emjin**
To track semgrep performance, run the small and medium benchmarks, with all variants, on each pull request merge (push to develop), where small is under 30 seconds and medium is under 3 minutes. We will then be able to plot semgrep perf changes over time

PR checklist:
- [ ] changelog is up to date

### CODE REVIEW

File: .github/workflows/upload-benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+name: Upload-benchmarks
+
+on:
+  workflow_dispatch:
+  pull_request:
```
Comment (aryx) [inline .github/workflows/upload-benchmarks.yml]: if we just want to do it on merge to develop, we should remove this line no? We should not do that on each PR?

File: .github/workflows/upload-benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+jobs:
+  build-core:
+    name: Build semgrep-core
+    runs-on: ubuntu-latest
+    container: returntocorp/ocaml:alpine
+    steps:
+      - name: Pre-checkout fixes
+        run: |
+          sudo chmod -R 777 /github
```
Comment (aryx) [inline .github/workflows/upload-benchmarks.yml:17]: what is this? Do we do that in the other workflow?

**Review by aryx — APPROVED** [review]

File: .github/workflows/upload-benchmarks.yml (hunk 1 above)
Comment (emjin) [reply .github/workflows/upload-benchmarks.yml]: That's just for testing---I'm actually going to change it to nightly

File: .github/workflows/upload-benchmarks.yml (hunk 2 above)
Comment (emjin) [reply .github/workflows/upload-benchmarks.yml:17]: Yes, this is copied from tests.yml, the build and test semgrep-core job.

Comment (aryx) [reply .github/workflows/upload-benchmarks.yml:17]: put a comment then, like coupling: this is copy-pasted from tests.yml#build and test semgrep-core job

File: perf/bench/big-js/prep [hunk 3]
```diff
@@ -36,6 +36,6 @@ shallow_clone() {
 name="big-js"
 #TODO: the other repo are using https: but this repo is currently private
 # so I use ssh
```
Comment (mjambon) [inline perf/bench/big-js/prep:38]: ^ remove the old comment

**Review by mjambon — COMMENTED** [review]
This is a lot of duplicated code that's going to be hard to maintain and slow to run. Compare with
https://github.com/returntocorp/semgrep/blob/develop/.circleci/config.yml
which uses the latest docker build from the develop branch. We should not have to rebuild semgrep each time we want to use it.

**Pull request closed without merging**